packages/core/src/Tool.ts

```diff
@@ -343,6 +346,30 @@ export class Tool extends Contract<ToolOptions> {
 		);
 	}
 
+	/**
+	 * Resolve modules on *behalf* of the configuration module and within the context
+	 * of its dependencies. This functionality is necessary to satisfy Yarn PnP and
+	 * resolving plugins that aren't listed as direct dependencies.
```

Comment:
NIT: It's required regardless of PnP, otherwise it's depending on hoisting which isn't guaranteed to be in its favour, e.g. workspaces.
https://yarnpkg.com/advanced/rulebook#packages-should-only-ever-require-what-they-formally-list-in-their-dependencies